docs/DeveloperGuide.md

```diff
@@ -1,14 +1,203 @@
+    - `keywords`: An array of expected keywords for argument extraction.
+    - `arguments`: The map where extracted arguments will be stored.
+    - `prevKeyword`: The keyword found in the previous recursive call.
+  - **Description**: Extracts values associated with keywords and updates the arguments map accordingly.
+  - **Process**:
+    - Base case: No argument left to split: `argumentString.isEmpty()`
+    - Find the first keyword in the list that appears in the argumentString
+    - If found:
+      - Attach the part before the keyword with the previously found keyword and put in to `arguments`.
+      - Delete the keyword from the `keywords` list (to not be considered in the next call)
+      - Pass the remaining `argumentString` after the keyword to the next recursive call
+    - If not found (mean that the last keyword reached):
+      - Attach the remaining part with the previously found keyword and put in to `arguments`.
+
+### Category
+The `Category` class encapsulates the name of a category and provides functionality for equality checks, hash code generation, and string representation. It serves as the foundational representation of a category.
+
+#### Class Responsibilities
+1. **Attribute Encapsulation**: Encapsulates the category name to prevent external modification.
+2. **Equality Checks**: Implements equality based on the category name, allowing categories with the same name to be considered equal.
+3. **String Representation**: Provides a `toString` method for convenient logging and debugging output.
+
+#### Class attributes
+1. **name**:String
+   - Description:Represents the category name, set as a read-only attribute.
+
+#### Class Main Methods
+1. **public Category(String name)**
+   - **Parameters**:
+     - **`name`**: The category name.
+   - **Process**: Initializes the name attribute with the specified value.
+
+2. **public String getName()**
+   - **Returns**: The `name` of the Category object.
+   - **Process**: Provides access to name attribute.
+
+3. **public boolean equals(Object obj)**
+   - **Parameters**:
+     - **`obj`**: The object to compare for equality.
+   - **Returns**: `true` if `obj` is a Category instance with the same name, `false` otherwise.
+   - **Process**: Checks if `obj` is a Category instance and compares its name with the current `Category` object’s name.
+
+4. **public int hashCode()**
+   - **Returns**: The hash code based on the name.
+   - **Process**: Calculates the hash code for storing `Category` objects in hash-based collections.
+
+5. **public int hashCode()**
+   - **Returns**: The formatted description of the category
+   - **Process**: Generates a string representation of the `Category` object, useful for logging and debugging.
+
+![Category](./diagrams/category/category-class-diagram.png)
+
+### TransactionList
+The `TransactionList` class is responsible for storing user transactions of different types. It also provides various
```

Comment:
In the class diagram, I feel that you could exclude the `get` functions, as their usage and implementation are quite trivial to the architecture of the class. 😅